### ISSUE TITLE
extended mode shouldn't finds numbers inside long strings 

### ISSUE THREAD

**Issue by hereismass** [catamphetamine/libphonenumber-js#213]
Hello,

`findPhoneNumbers` in extended mode finds numbers inside big strings like UUID for example.
These numbers can then be recognized as valid:

```javascript
const text = 'CA801c26f98cd16e231354125ad046e40b';
const numbers = findPhoneNumbers(text, 'FR', { extended: true });
/* numbers = [{
  phone: "801",
  valid: false,
  ...
},{
  phone: "231354125",
  valid: true,
  ...
},{
  phone: "046",
  valid: false,
  ...
}]
```

This doesn't happen with `parseNumber` in extended, nor the google demo page (which I guess is the parseNumber function also)

thanks!

**Comment by catamphetamine** [comment]
> nor the google demo page (which I guess is the parseNumber function also)

Google's original phone number search is not used in this library (because it's not ported).
Instead a custom `findPhoneNumbers()` function was written.
I looked at Google's code and saw how they filter out those matches where phone numbers are surrounded by letters.
So I added the same code to `findPhoneNumbers()`.
Released `libphonenumber-js@1.2.10`.
